api/storage_service.cc

```diff
@@ -717,15 +717,14 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
     ss::remove_node.set(r, [&ss](std::unique_ptr<request> req) {
         auto host_id = req->get_query_param("host_id");
         std::vector<sstring> ignore_nodes_strs= split(req->get_query_param("ignore_nodes"), ",");
-        auto ignore_nodes = std::list<gms::inet_address>();
+        auto ignore_nodes = std::list<sstring>();
```

Comment:
I understand why you convert to string here. But it is sad. Can we do the full parse of  the ignore_nodes here?

Reply:
Because we know the host_id in the string is for sure host_id. But before this patch, we pass std::list<gms::inet_address> to ss:removenode(). After, we pass a std::list<sstring>. This is a step back.  We lose the type info. If after this patch, we must to do the parse in removenode, let's do all the parse there, i.e., moving the parse from ss::remove_node.set to ss:removenode. Not part of the parse in api and part in stroage_service.

Reply:
We could verify the host id is at least a uuid in the api layer. 
But see my comments above, move all the parse stroage_sercie:: removenode().  

Do you agree it is better to do parse in one place?

Reply:
In general, I think the api layer should be responsible for the superficial parsing and validating the syntax, rejecting invalid arguments, while the semantic validation should be done at the service layer.
I.e. it's out of scope of the service::storage_service layer to make sure the list of nodes to ignore is all UUIDs or all ip addresses - that's the api's business.

We can change the storage_service api to get strongly typed parameters and do further validation based on the token_metadata, but I think it's a hassle, and in this case it's fine to pass strings down the stack.

In the next version, I'll some basic validation at the api layer so the strings are at least validated from the syntax perspective.